del_timer_sync is undefined

Kernel commit https://github.com/torvalds/linux/commit/8fa7292fee5c5240402371ea89ab285ec856c916 removed the wrapper for `del_timer_sync`, which now causes compilation to fail on kernels >= 6.15:

```
wctc4xxp/base.c: In function 'wctc4xxp_remove_one':
wctc4xxp/base.c:4168:13: error: implicit declaration of function 'del_timer_sync'; did you mean 'dev_mc_sync'? [-Werror=implicit-function-declaration]
 4168 |         if (del_timer_sync(&wc->watchdog))
      |             ^~~~~~~~~~~~~~
      |             dev_mc_sync
cc1: all warnings being treated as errors
make[5]: *** [/usr/src/linux/scripts/Makefile.build:203: wctc4xxp/base.o] Error 1
make[4]: *** [/usr/src/linux/scripts/Makefile.build:461: wctc4xxp] Error 2
make[3]: *** [/usr/src/linux/Makefile:2003: .] Error 2
make[2]: *** [/usr/src/linux/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/usr/src/dahdi-linux-3.4.0/drivers/dahdi'
make[1]: *** [Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/usr/src/linux'
make: *** [Makefile:74: modules] Error 2
```

--- a/include/dahdi/kernel.h
+++ b/include/dahdi/kernel.h
@@ -66,6 +66,11 @@
 #endif /* RHEL_RELEASE_CODE */
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,15,0)
+#define del_timer timer_delete
+#define del_timer_sync timer_delete_sync
+#endif
+
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 18, 0)
 #include <linux/pci.h>
 #include <linux/dma-mapping.h>

